lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c

```diff
@@ -113,52 +118,120 @@ void bit_rate_value_to_eps_qos(
+        ambr_dl_kbps - (apn_ambr->apnambrfordownlink_extended2) * 256000;
     apn_ambr->apnambrfordownlink = 0xff;
-  } else if ((ambr_dl_kbps > 0) && (ambr_dl_kbps <= 63)) {
-    apn_ambr->apnambrfordownlink = ambr_dl_kbps;
-  } else if ((ambr_dl_kbps > 63) && (ambr_dl_kbps <= 575)) {
-    apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 64) / 8) + 64;
-  } else if ((ambr_dl_kbps > 575) && (ambr_dl_kbps <= 8640)) {
-    apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 576) / 64) + 128;
-  } else if (ambr_dl_kbps > 8640) {
-    apn_ambr->apnambrfordownlink = 0xfe;
-    apn_ambr->extensions =
+  }
+
+  if ((ambr_dl_kbps > 8640) &&
+      ((ambr_dl_kbps > 8600) && (ambr_dl_kbps <= 16000))) {
+    apn_ambr->extensions |=
+        APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
+    apn_ambr->apnambrfordownlink_extended = (ambr_dl_kbps - 8600) / 100;
+
+    ambr_dl_kbps =
+        ambr_dl_kbps - (apn_ambr->apnambrfordownlink_extended) * 100 - 8600;
+  }
+
+  else if ((ambr_dl_kbps > 16000) && (ambr_dl_kbps <= 128000)) {
+    apn_ambr->extensions |=
+        APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
+    apn_ambr->apnambrfordownlink_extended =
+        ((ambr_dl_kbps - 16000) / 1000) + 74;
+    ambr_dl_kbps = ambr_dl_kbps -
+                   (apn_ambr->apnambrfordownlink_extended - 74) * 1000 - 16000;
+
+  } else if ((ambr_dl_kbps > 128000) && (ambr_dl_kbps <= 256000)) {
+    apn_ambr->extensions |=
         APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
-    if ((ambr_dl_kbps >= 8600) && (ambr_dl_kbps <= 16000)) {
-      apn_ambr->apnambrfordownlink_extended = (ambr_dl_kbps - 8600) / 100;
-    } else if ((ambr_dl_kbps > 16000) && (ambr_dl_kbps <= 128000)) {
-      apn_ambr->apnambrfordownlink_extended =
-          ((ambr_dl_kbps - 16000) / 1000) + 74;
-    } else if ((ambr_dl_kbps > 128000) && (ambr_dl_kbps <= 256000)) {
-      apn_ambr->apnambrfordownlink_extended =
-          ((ambr_dl_kbps - 128000) / 2000) + 186;
+    apn_ambr->apnambrfordownlink_extended =
+        ((ambr_dl_kbps - 128000) / 2000) + 186;
+
+    ambr_dl_kbps = ambr_dl_kbps -
+                   (apn_ambr->apnambrfordownlink_extended - 186) * 2000 -
+                   128000;
+  }
+
+  if ((apn_ambr->apnambrfordownlink != 0xfe) && !(ambr_dl_kbps > 8640)) {
+    if (ambr_dl_kbps == 0) {
+      apn_ambr->apnambrfordownlink = 0xff;
+    }
+    if ((ambr_dl_kbps > 0) && (ambr_dl_kbps <= 63)) {
+      apn_ambr->apnambrfordownlink = ambr_dl_kbps;
+    } else if ((ambr_dl_kbps > 63) && (ambr_dl_kbps <= 575)) {
+      apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 64) / 8) + 64;
+      ambr_dl_kbps =
+          ambr_dl_kbps - (apn_ambr->apnambrfordownlink_extended - 64) * 8 - 63;
+    } else if ((ambr_dl_kbps > 575) && (ambr_dl_kbps <= 8640)) {
+      apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 576) / 64) + 128;
+      ambr_dl_kbps                 = ambr_dl_kbps -
+                     (apn_ambr->apnambrfordownlink_extended - 128) * 64 - 575;
```

Comment:
please run `make build_oai` before adding files and committing to fix clang formatting issues.

Reply:
Ran make format_core to format the code.